go-controller/pkg/ovn/controller/services/load_balancer.go

```diff
@@ -56,23 +59,27 @@ var protos = []v1.Protocol{
 // - services with NodePort set
 // - services with host-network endpoints
 // - services with ExternalTrafficPolicy=Local
+// - services with InternalTrafficPolicy=Local
 func buildServiceLBConfigs(service *v1.Service, endpointSlices []*discovery.EndpointSlice) (perNodeConfigs []lbConfig, clusterConfigs []lbConfig) {
 	// For each svcPort, determine if it will be applied per-node or cluster-wide
 	for _, svcPort := range service.Spec.Ports {
 		eps := util.GetLbEndpoints(endpointSlices, svcPort)
 
-		// if ExternalTrafficPolicy is local, then we need to do things a bit differently
+		// if ExternalTrafficPolicy or InternalTrafficPolicy is local, then we need to do things a bit differently
 		externalTrafficLocal := (service.Spec.ExternalTrafficPolicy == v1.ServiceExternalTrafficPolicyTypeLocal)
+		internalTrafficLocal := (*service.Spec.InternalTrafficPolicy == v1.ServiceInternalTrafficPolicyLocal)
```

Comment:
Do we need a nil check here?

Reply:
oh good point, better to be safe than sorry with pointers,